src/main/java/com/savvato/tribeapp/services/PhraseServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.savvato.tribeapp.services;
+
+import com.savvato.tribeapp.dto.PhraseDTO;
+import com.savvato.tribeapp.entities.Phrase;
+import com.savvato.tribeapp.repositories.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PhraseServiceImpl implements PhraseService {
+
+    @Autowired
+    UserPhraseService userPhraseService;
+
+    @Autowired
+    PhraseRepository phraseRepository;
+
+    @Autowired
+    AdverbRepository adverbRepository;
+
+    @Autowired
+    NounRepository nounRepository;
+
+    @Autowired
+    PrepositionRepository prepositionRepository;
+
+    @Autowired
+    VerbRepository verbRepository;
+
+    @Override
+    public Optional<List<PhraseDTO>> getListOfPhraseDTOByUserId(Long userId) {
+
+        // create list of PhraseDtos
+        List<PhraseDTO> phraseDTOS = new ArrayList<>();
+
+        // Create list of phrase ids
+        Optional<List<Long>> optPhraseIds = userPhraseService.findPhraseIdsByUserId(userId);
+
+        // Create list of all phrases from those phrase ids
+        if (optPhraseIds.isPresent()) {
+            List<Long> phraseIds = optPhraseIds.get();
+            List<Phrase> phrases = new ArrayList<>();
+
+            // loop through list of phrase ids and add each phrase to list
+            for (Long phraseId : phraseIds) {
+                Optional<Phrase> optPhrase = phraseRepository.findPhraseByPhraseId(phraseId);
+                if (optPhrase.isPresent()) {
+                    phrases.add(optPhrase.get());
+                }
+            }
+
+            // loop through phrases and get words
+            for (Phrase phrase : phrases) { //do we need to check again if empty
+                PhraseDTO phraseDTO = PhraseDTO.builder().build();
+
+                Optional<String> optAdverb = adverbRepository.findAdverbById(phrase.getAdverbId());
+                Optional<String> optVerb = verbRepository.findVerbById(phrase.getVerbId());
+                Optional<String> optPreposition = prepositionRepository.findPrepositionById(phrase.getPrepositionId());
+                Optional<String> optNoun = nounRepository.findNounById(phrase.getNounId());
+
+                if (optAdverb.isPresent()) {
```

Comment:
You can assume the word objects are indeed present, no need to check isPresent() before calling .get(). 

This is because if they are not present, that means our data is out of wack and we want to know about that via the errors and exceptions it will cause.

Reply:
Should I leave it in place for the optional words? If I remember correctly, weren't the verb and noun required, but the adverb and preposition are optional (or something like that)?I thought I remembered hearing that in one of our meetings, but correct me if I'm crazy.